completions/bash/clr-installer

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# shellcheck disable=SC2207
+# -----------------------------------------------------------------------
+#   Clear Linux OS* Installer - autocompletion script
+#
+#   Author: Lucius Hu - http://github.com/lebensterben
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, version 2 or later of the License.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------
+
+_clr_installer() {
+  local IFS opts prev cur
+
+  _init_completion -s || return
+
+  case "$prev" in
+    --archive|--copy-network|--iso|--keep-image|--reboot|--stub-image|--swupd-skip-diskspace-check|--telemetry)
+      COMPREPLY=($(compgen -W "true false" -- "$cur"))
+      return
+      ;;
+    --block-device)
+      local blockdevice; blockdevice="$(lsblk --noheadings --raw --paths | cut -d' ' -f1 | tr '\n' ',')"
+      # The trick is to insert any UNICODE space(s), e.g. non-breaking space, before the message
+      #    so that 'word' is not likely to be matched by a user's input
+      # And add at least one other 'word' start with any UNICODE space(s)
+      #    otherwise completion system will push the only candidate to the command line
+      # The drawback is, this special character may not shown correctly, for example in `tty`, it's
+      #    it's shown as a rectangle
+
+      # here, before F and B, there is a UNICODE tag space U+E0020
+      opts="󠀠Format:\"alias1:filename1[,alias2:filename2,...]\" 󠀠BlockDevices:${blockdevice:0:-1}"
+      COMPREPLY=($(compgen -W "$opts" -- "$cur"))
+      return
+      ;;
```

Comment:
I wonder if it is reasonable or worthwhile to add --bundle with a completion similar to what swupd as?  See /usr/share/defaults/etc/profile.d/50-swupd.bash
The list would probably get too long?

Reply:
The list is getting from MoM file.
Does installer live ISO has the MoM file as well? (And is it in same location?)

Reply:
The installer has a MoM file, but the `/var/lib/swupd/version` referenced in the code for swupd does not exist. We've be better to use `swupd info --quiet` to get the installed version.

This was just a nice-to-have. Do not worry about this one, we can add it later.

Reply:
`swupd info | sed '2q;d' | cut -d' ' -f3` can get the version string.
Can you help to verify that `sed` is available in the live ISO (both server and desktop)?

Reply:
```
swupd search /usr/bin/sed
Bundle with the best search result:
     os-core                        	- Run a minimal Linux userspace.  (64MB) (installed)
```
sed is is os-core, so it will always be present

Reply:
> `swupd info | sed '2q;d' | cut -d' ' -f3` can get the version string.
> Can you help to verify that `sed` is available in the live ISO (both server and desktop)?

Why add the two pipes, sed, and cut if `swupd info --quiet` does the job?

Reply:
I just burned a live iso with latest version.

Unfortunately,
- Running `swupd` gives `Input/output error`. So `swupd info --version`
  won't work
- There's no `/var/lib/swupd`.

Can you reproduce this?

Reply:
Okay, let's drop this from the PR.
The content is there in the ISO I generate when testing the installer; Dev Ops must be using the clean swupd option to reduce the image time.